### ISSUE TITLE
(void-function dired-create-files) error when trying to copy file through helm-find-file actions.

### ISSUE THREAD

**Issue by vibrys** [syl20bnr/spacemacs#9632]
#### Description :octocat:
(void-function dired-create-files) error when trying to copy file through helm-find-file actions.

#### Reproduction guide :beetle:
- Start Emacs
- SPC f f
- choose any file, for example ~/.Xauthority
- visit helm-find-file actions
- choose "copy File(s)" action
- as destination file, choose for example /sudo::/tmp/zz.txt

*Observed behaviour:* :eyes: :broken_heart:
(void-function dired-create-files)

*Expected behaviour:* :heart: :smile:
file should be copied.

#### System Info :computer:
- OS: gnu/linux
- Emacs: 25.3.1
- Spacemacs: 0.200.9
- Spacemacs branch: nil (rev. ca28bde)
- Graphic display: t
- Distribution: spacemacs
- Editing style: emacs
- Completion: helm
- Layers:
```elisp
(helm emacs-lisp)
```
- System configuration features: XPM JPEG TIFF GIF PNG RSVG SOUND GPM DBUS GCONF GSETTINGS NOTIFY GNUTLS LIBXML2 FREETYPE M17N_FLT LIBOTF XFT ZLIB TOOLKIT_SCROLL_BARS GTK2 X11

#### Backtrace :paw_prints:
```
Debugger entered--Lisp error: (void-function dired-create-files)
  dired-create-files(dired-copy-file "copy" ("/home/user/.Xauthority") #[257 "\300\207" ["/sudo:root@host:/tmp/zz.txt"] 2 "\n\n(fn FROM)"] 67)
  helm-dired-action("/sudo:root@host:/tmp/zz.txt" :files ("/home/user/.Xauthority") :action copy :follow nil)
  helm-find-files-do-action(copy)
  helm-find-files-copy("/home/user/.Xauthority")
  helm-execute-selection-action-1()
  helm-execute-selection-action()
  helm-internal(helm-source-find-files "/home/user/" "Find files or url: " nil nil "*helm find files*" nil nil nil)
  apply(helm-internal (helm-source-find-files "/home/user/" "Find files or url: " nil nil "*helm find files*" nil nil nil))
  helm(helm-source-find-files "/home/user/" "Find files or url: " nil nil "*helm find files*" nil nil nil)
  apply(helm (helm-source-find-files "/home/user/" "Find files or url: " nil nil "*helm find files*" nil nil nil))
  helm(:sources helm-source-find-files :input "/home/user/" :case-fold-search smart :preselect nil :ff-transformer-show-only-basename t :default nil :prompt "Find files or url: " :buffer "*helm find files*")
  helm-find-files-1("/home/user/")
  (let* ((hist (and arg helm-ff-history (helm-find-files-history))) (default-input hist) (input (cond ((and (eq major-mode (quote dired-mode)) default-input) (file-name-directory default-input)) ((and (not (string= default-input "")) default-input)) (t (expand-file-name (helm-current-directory)))))) (set-text-properties 0 (length input) nil input) (helm-find-files-1 input))
  spacemacs/helm-find-files(nil)
  funcall-interactively(spacemacs/helm-find-files nil)
  call-interactively(spacemacs/helm-find-files nil nil)
  command-execute(spacemacs/helm-find-files)
```

**Comment by vibrys** [comment]
I've just installed vanilla helm (WITHOUT any other packages) on my emacs and it turns out that this error also persist there, thus this does NOT seem to be spacemacs one.

I'm very sorry for the noise - I cannot close this defect.

best regards,
Mat

**Comment by bmag** [comment]
Reported upstream: https://github.com/emacs-helm/helm/issues/1878 (@vibrys why didn't you link that issue here?)

Maybe fixed upstream: https://github.com/emacs-helm/helm/commit/e9b8366211fe5d4b3923cfa106522db3f92d1d21 (though I guess it isn't, cause `dired-create-files` is defined in `dired-aux.el` and that commit just requires `dired.el`)

**Comment by vibrys** [comment]
@bmag, I didn't refrence it, because that error was NOT the exact problem I experienced here. Problem https://github.com/emacs-helm/helm/issues/1878 has been fixed and as a consequence it reveals THIS problem, but is up to thierryvolpiatto if I should file another issue within helm to address it. If not, then it is the moment to reference that from here.

**Comment by bmag** [comment]
So let me just understand the timeline:
- you opened this issue here, and almost immediately commented that it's an upstream problem
- 2 hours later you opened an upstream issue about a similar but different error
- another 8 hours later that issue is fixed upstream
- that fix reveals the error you reported here earlier

That doesn't add up, but whatever. This issue should also be fixed now (see https://github.com/emacs-helm/helm/commit/b0be214cf64526c686ae823da4709d647b59a2ad)

**Comment by vibrys** [comment]
just after opening the issue here, I realized it is better to check it on vanilla emacs+helm only setup, but it lead to another problem. I did fast, but dirty  fix at my desktop for it, then the above error appeared.

I've decided not to report two errors within the same issue, but instead wait on helm team to fix the first issue and then ask what to do with another issue (open new one?).

Opening new issue seems to be a bit attractive, because people search for bugs they meet within the set of already closed issues. Having it within a title outlines the problem is already closed.
